src/main/java/io/github/thebusybiscuit/slimefun4/core/services/CustomItemDataService.java

```diff
@@ -45,6 +45,9 @@ public void setItemData(ItemMeta im, String id) {
     }
 
     public Optional<String> getItemData(ItemStack item) {
+        if (item.getItemMeta() == null) {
```

Comment:
You should do .hasItemMeta() here (and maybe also add a null-check for the given ItemStack parameter as pointed out before), otherwise you won't gain any performance benefit from this here, since #getItemMeta() (for normal ItemStacks) will create a new instance of ItemMeta and perform a clone operation which is very costly.